frontend/components/Course/CourseSettings/CourseForm.tsx

```diff
@@ -138,6 +168,76 @@ const CourseForm = (props: CourseFormProps) => {
             });
     };
 
+    useEffect(() => {
```

Comment:
I don't think this is actually getting called when you expect it to be - tags information are not part of the course object. You will have to probably move this logic into onSubmit. 

A side effect of this - I think tags are pretty buggy on the course settings page now. Try interacting with the list after saving once.